shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/core/api/impl/RuleAlteredJobAPIImpl.java

```diff
@@ -322,10 +322,7 @@ public void switchClusterConfiguration(final JobConfiguration jobConfig) {
         String ruleCacheId = workflowConfig.getRuleCacheId();
```

Comment:
Looks `JobStatus.ALMOST_FINISHED` still might be not set, since `optionalJobContexts` might be empty as discussed before.